node/src/tls.rs

```diff
@@ -738,7 +740,7 @@ impl PartialOrd for Sha512 {
 
 impl Debug for Sha512 {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", HexFmt(&self.0[..]))
+        write!(f, "{}", checksummed_hex::encode(&self.0[..]))
```

Comment:
```suggestion
        write!(f, "{}", HexFmt(&self.0[..]))
```

Sha512s are so long that `checksummed_hex::encode` will be lower case.

We should perhaps consider reverting all of the places that we got rid of `HexFmt` and `HexList`...

Reply:
> Sha512s are so long that checksummed_hex::encode will be lower case.

I don't think that's correct.  We use checksumming for inputs <= 75 bytes, and SHA512 digests are 64 bytes.  I'll make the specific `HexFmt`/`HexList` changes you requested in the rest of this review, but leave the others as is.